lib/services/keyVault/lib/keyvault.d.ts

```diff
@@ -42,7 +42,7 @@ export interface Authenticator {
  * @param {KeyVaultCredentials~authRequest} authenticator  A callback that receives a challenge and returns an authentication token.
  */
 export class KeyVaultCredentials implements msRest.ServiceClientCredentials {
-  cosntructor(authenticator: Authenticator);
+  constructor( authenticator:  (challenge: any, callback: any) => any );
```

Comment:
while I agree with the constructor typo. Why would the Authenticator type not work? `constructor(authenticator: Authenticator);` Is there a reason to make challenge and callback of type "any"?
We have exposed the interface Authenticator [here](https://github.com/Azure/azure-sdk-for-node/pull/2136/files#diff-909a9a4b181a56c07f397c89bd074be5R31).
/cc @balajikris - We need to ship a patch version (with constructor fix) irrespective of the TS samples.